api/cloud_api/util/SceGoogleApiHandler.js

```diff
@@ -141,23 +144,27 @@ class SceGoogleApiHandler {
    */
   getEventsFromCalendar(calendarId, numOfEvents) {
     return new Promise((resolve, reject) => {
-      const calendar =
-        google.calendar({ version: 'v3', auth: this.oAuth2Client });
-      calendar.events.list({
-        calendarId: calendarId,
-        timeMin: (new Date()).toISOString(),
-        maxResults: numOfEvents,
-        singleEvents: true,
-        orderBy: 'startTime',
-      }, (err, res) => {
-        if (err) return reject(false);
-        const events = res.data.items;
-        if (events.length) {
-          resolve(events);
-        } else {
-          reject(false);
-        }
-      });
+      if(this.hasValidAPIKeys) {
```

Comment:
instead of wrapping the functions in an if/else, at the top of each you can do

```js
if (!this.hasValidAPIKeys) {
  return resolve(true);
}
// rest of original function
```